CollectionOfHelpers/CollectionOfHelpers/Specialised/Sudoku/SudokuDistributedGridCollection.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace CollectionOfHelpers.Specialised.Sudoku
+{
+    public class SudokuDistributedGridCollection
+    {
+        public int GridSize => Columns.Length;
+        private int boxSize;
+
+        private SudokuSubCollection[] Columns;
+        private SudokuSubCollection[] Rows;
+        private SudokuSubCollection[] Boxes;
+
+        public SudokuDistributedGridCollection(int gridSize)
+        {
+            throwExceptionForInvalidGridSize(gridSize);
+            boxSize = (int) Math.Sqrt(gridSize);
+            initialiseGrid(gridSize);
+        }
+
+        private void throwExceptionForInvalidGridSize(int gridSize)
```

Comment:
A fair point. However, I was trying to be explicit as simply ValidateGridSize doesn't tell you what sort of action will be taken when the grid is invalid - for example will it be corrected, will a warning be displayed, etc.
We can discuss the pros and cons in person as it is an interesting case - explicit names vs conventions vs easy future modifications.